Make short url instance available in middleware

moritzlang opened the pull request:
Hey @ash-jc-allen,

First of all: thanks for your work on this package!
The background for this PR is, that I need to dynamically add query parameters on the destination url. I also need to access the `ShortURL` instance of the link for this.

Although this can already be done by adding a middleware through the `middleware` config option, I would need to query the `ShortURL` instance which is later retrieved again in the `ShortURLController` [here](https://github.com/ash-jc-allen/short-url/blob/664ee84d9df81de7617aec30643e71f69ce79ea3/src/Controllers/ShortURLController.php#L24). In order to prevent executing the same query twice I propose this PR. Furthermore having access to this current `ShortURL` instance makes middlewares way more powerful.

ash-jc-allen commented:
Hey @moritzlang, thanks for the PR! 😄

I do kind of like the idea of what you're trying to achieve with these changes. But I'm wondering if there's a different way to add this functionality without the breaking change (i.e. - not changing the controller's method signature)?

I'd like to try and avoid needing to do a major release for the package so I can keep the upgrade path nice and simple.

I think I'll probably close this PR for the time being. But if it's something that enough people want and would find useful, I'll definitely make sure to revisit this!

Thanks for taking the time to make the PR though, I appreciate it! 🙂

moritzlang commented:
Thanks for the feedback @ash-jc-allen!
I added a way that gracefully handles the existing controller method signature and fixed a type error that made the tests fail 😄. Would be great if you could have another look at it.
